scripts/src/java/org/oppia/android/scripts/coverage/CoverageReporter.kt

```diff
@@ -419,22 +434,31 @@ class CoverageReporter(
           totalLinesHit, totalLinesFound
         )
 
-        val exemption = testFileExemptionList[filePath]
-        if (coveragePercentage < MIN_THRESHOLD) {
-          if (exemption != null) {
-            val ovveriddenMinCoverage = exemption.overrideMinCoveragePercentRequired
-            if (coveragePercentage < ovveriddenMinCoverage) {
-              return CoverageCheck.FAIL
-            }
-          } else {
-            return CoverageCheck.FAIL
-          }
-        }
+        val threshold = testFileExemptionList[filePath]
+          ?.overrideMinCoveragePercentRequired
+          ?: MIN_THRESHOLD
+        if (coveragePercentage < threshold) return CoverageCheck.FAIL
       }
     }
     return CoverageCheck.PASS
   }
 
+  private fun calculateOverallCoveragePercentage(): Float {
+    var totalLinesFound = 0
+    var totalLinesHit = 0
+
+    coverageReportContainer.coverageReportList.forEach { report ->
+      report.details?.let {
+        totalLinesFound += it.linesFound
+        totalLinesHit += it.linesHit
+      }
+    }
+
+    return totalLinesFound.takeIf { it > 0 }
+      ?.let { totalLinesHit.toFloat() / it * 100 }
+      ?: 0.0f
+  }
```

Comment:
```suggestion
    val reports = coverageReportContainer.coverageReportList
    val totalLinesFound = reports.sumOf { it.linesFound }.toFloat()
    val totalLinesHit = report.sumOf { it.linesHit }.toFloat()
    return if (totalLinesFound > 0) (totalLinesHit * 100.0f) / totalLinesFound else 0.0f
```

More compact and idiomatic (no need to mess with mutable state).

Reply:
Ah! Got it, updated with the same logic.